File: solutions/day-21-subarray-sum-equals-k/Solution.java
import java.util.*;

class Solution {
    public int subarraySum(int[] nums, int k) {
        int count = 0;
        int prefix = 0;
        Map<Integer, Integer> seen = new HashMap<>();
        seen.put(0, 1);

        for (int num : nums) {
            prefix += num;
            if (seen.containsKey(prefix - k)) {
                count += seen.get(prefix - k);
            }
            seen.put(prefix, seen.getOrDefault(prefix, 0) + 1);
        }

        return count;
    }
}

File: solutions/day-21-subarray-sum-equals-k/Solution.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    int subarraySum(std::vector<int>& nums, int k) {
        int count = 0;
        int prefix = 0;
        std::unordered_map<int, int> seen;
        seen[0] = 1;

        for (int num : nums) {
            prefix += num;
            if (seen.count(prefix - k)) {
                count += seen[prefix - k];
            }
            seen[prefix]++;
        }

        return count;
    }
};
